src/com/example/Demo_Subj/RoomActivity.java

```diff
@@ -106,8 +86,8 @@ public void onCreate(Bundle savedInstanceState) {
         timer.schedule(new TimerTask(){
             @Override
             public void run(){
-                subject.tick();
-                GlobalInformation.setCurrentRoom(subject.getAktRoomID());
+                //subject.tick();
+                GlobalInformation.setCurrentRoom(grafik.getAktRoomID());
```

Comment:
Auch hier: Das Subjekt entscheidet logisch darüber, in welchem Raum es ist, nicht die Grafikausgabe.